api/v1/compliance_report_types.go

```diff
@@ -0,0 +1,159 @@
+// Copyright (c) 2024 Tigera, Inc. All rights reserved.
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+)
+
+// ComplianceReportPodTemplate is the configuration for the ComplianceReport PodTemplate.
+type ComplianceReportPodTemplate struct {
+
+	// Spec is the specification of the ComplianceReport PodTemplateSpec.
+	// +optional
+	Template *ComplianceReportPodTemplateSpec `json:"spec,omitempty"`
+}
+
+// ComplianceReportPodTemplateSpec is the ComplianceReport PodTemplateSpec.
+type ComplianceReportPodTemplateSpec struct {
+
+	// Spec is the ComplianceReport PodTemplate's PodSpec.
+	// +optional
+	Spec *ComplianceReportPodSpec `json:"spec,omitempty"`
+}
+
+// ComplianceReportPodSpec is the ComplianceReport PodSpec.
+type ComplianceReportPodSpec struct {
+	// InitContainers is a list of ComplianceReport PodSpec init containers.
+	// If specified, this overrides the specified ComplianceReport PodSpec init containers.
+	// If omitted, the ComplianceServer Deployment will use its default values for its init containers.
+	// +optional
+	InitContainers []ComplianceReportPodTemplateInitContainer `json:"initContainers,omitempty"`
+
+	// Containers is a list of ComplianceServer containers.
+	// If specified, this overrides the specified ComplianceReport PodSpec containers.
+	// If omitted, the ComplianceServer Deployment will use its default values for its containers.
+	// +optional
+	Containers []ComplianceReportPodTemplateContainer `json:"containers,omitempty"`
+}
+
+// ComplianceReportPodTemplateContainer is a ComplianceServer Deployment container.
+type ComplianceReportPodTemplateContainer struct {
+	// Name is an enum which identifies the ComplianceServer Deployment container by name.
+	// +kubebuilder:validation:Enum=reporter
```

Comment:
Could you rename the container and this value from "reporter" to "compliance-reporter", to make it consistent with the others?

Reply:
compliance-controller have a changes that looks for the container named "reporter" for compliance reports..
https://github.com/tigera/calico-private/blob/54ba24b781a2d62c737ac4a25466acd8a619e196/compliance/pkg/controller/controller.go#L498
After renaming the `reporter` container name  to `compliance-reporter`, reports were not getting generated.. 
During upgrade, controller would have already configured jobs with  container name "reporter", after upgrade compliance -controller would be modified to look for "compliance-reporter" container hence it would produce any report. i guess it might be breaking changes. unless we delete all the jobs and create manually again.

WDYT?

Reply:
After discussing this offline, we decided that the breaking changes for upgrades would outweigh the benefits of my suggestion.